vsintegration/src/FSharp.Editor/Hints/RoslynAdapter.fs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor.Hints
+
+open System.Collections.Immutable
+open System.ComponentModel.Composition
+open Microsoft.CodeAnalysis.ExternalAccess.FSharp.InlineHints
+open Microsoft.VisualStudio.FSharp.Editor
+
+// So the Roslyn interface is called IFSharpInlineHintsService
+// but our implementation is called just HintsService.
+// That's because we'll likely use this API for things other than inline hints,
+// e.g. signature hints above the line, pipeline hints on the side and so on.
+
+[<Export(typeof<IFSharpInlineHintsService>)>]
+type internal RoslynAdapter 
+    [<ImportingConstructor>]
+    (settings: EditorOptions) =
+
+    static let userOpName = "Hints"
+
+    interface IFSharpInlineHintsService with
+        member _.GetInlineHintsAsync(document, _, cancellationToken) =
+            task {
```

Comment:
Should be very careful with tasks in VS, we need to use JoinableTaskFactory if we go back to UI thread. Otherwise we can easily get a deadlock.

I suggest using the `async` here, as in all other places, and have a separate issue for rewriting VS part from async to task state machine.